### PR TITLE
SPARKC-598 return proper name of the catalog

### PR DESCRIPTION
**Pull request by jtgrabowski**
The CatalogPlugin requires the actual name in name() as it is
used in matching with spark.sql.catalog.<name> setting.

### CODE REVIEW

File: connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogNamespaceSpec.scala [hunk 1]
```diff
@@ -7,7 +7,7 @@ import scala.collection.JavaConverters._
 class CassandraCatalogNamespaceSpec extends CassandraCatalogSpecBase {
 
   "A Cassandra Catalog Namespace Support" should "initialize successfully" in {
-    spark.sessionState.catalogManager.currentCatalog.name() should include("Catalog cassandra")
+    spark.sessionState.catalogManager.currentCatalog.name() should be(catalogName)
```
Comment (RussellSpitzer) [inline connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogNamespaceSpec.scala:10]: I may or may not have ... added this test in a ton of other catalog specs

[info]   "cassandra" did not include substring "Catalog cassandra" (CassandraCatalogTableReadSpec.scala:22)
2292
You'll probably need to hit them all

Comment (RussellSpitzer) [reply connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogNamespaceSpec.scala:10]: Other than that, Patch lgtm :)

**Review by RussellSpitzer — APPROVED** [review]
There are a few more tests with the "should init" that check the nam that need to be fixed, but 👍  other than that

File: connector/src/main/scala/com/datastax/spark/connector/datasource/CassandraCatalog.scala [hunk 2]
```diff
@@ -440,7 +440,7 @@ object CassandraCatalog {
     * is used to quickly bail out if the namespace is not compatible with Cassandra
     */
   def checkNamespace(namespace: Array[String]): Unit = {
-    if (namespace.length != 1) throw new NoSuchNamespaceException(s"$OnlyOneNamespace : $namespace")
+    if (namespace.length != 1) throw new NoSuchNamespaceException(s"""$OnlyOneNamespace: ${namespace.mkString(".")}""")
```
Comment (RussellSpitzer) [inline connector/src/main/scala/com/datastax/spark/connector/datasource/CassandraCatalog.scala:443]: Unrelated?

Comment (jtgrabowski) [reply connector/src/main/scala/com/datastax/spark/connector/datasource/CassandraCatalog.scala:443]: yeah, the original version was not human readable

File: connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogNamespaceSpec.scala (hunk 1 above)
Comment (jtgrabowski) [reply connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogNamespaceSpec.scala]: indeed! thanks, fixed.
